src/components/ec-modal/ec-modal.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <transition name="ec-modal--fade">
+    <div
+      v-if="showModal"
+      class="ec-modal"
+      @click.self="closeModal()"
+    >
+      <div
+        class="ec-modal__content"
+        :class="{'ec-modal--lg': large}"
+      >
+        <header class="ec-modal__header">
+          <slot name="header" />
+          <ec-icon
+            v-if="isClosable"
+            class="ec-modal__close"
+            name="simple-close"
+            :size="24"
+            @click.stop="closeModal()"
+          />
+        </header>
+
+        <main class="ec-modal__main">
+          <slot name="main" />
+        </main>
+
+        <footer class="ec-modal__footer">
+          <div
+            v-if="showFooterLeftContent"
+            class="ec-modal__footer-left-content"
+          >
+            <slot name="footer-left-content" />
+          </div>
+
+          <button
+            v-if="this.$scopedSlots.negative"
+            class="ec-modal__negative-btn ec-btn ec-btn--md ec-btn--secondary ec-btn--rounded ec-modal__negative-btn--right"
+            @click="negativeAction()"
+          >
+            <slot name="negative" />
+          </button>
+
+          <button
+            v-if="this.$scopedSlots.positive"
+            :class="{'ec-modal__positive-btn--right': !this.$scopedSlots.negative}"
+            class="ec-modal__positive-btn ec-btn ec-btn--md ec-btn--primary ec-btn--rounded"
+            @click="positiveAction()"
+          >
+            <slot name="positive" />
+          </button>
+        </footer>
+      </div>
+
+    </div>
+  </transition>
+</template>
+
+<script>
+import EcIcon from '../ec-icon';
+
+export default {
+  components: {
+    EcIcon,
+  },
+  model: {
+    prop: 'showModal',
+    event: 'close',
+  },
+  props: {
+    showModal: {
+      type: Boolean,
+      default: false,
+    },
+    showFooterLeftContent: {
+      type: Boolean,
+      default: false,
+    },
+    isClosable: {
+      type: Boolean,
+      default: true,
+    },
+    large: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  created() {
+    const thisM = this;
+
+    document.addEventListener('keyup', (e) => {
+      if (e.keyCode === 27) {
```

Comment:
I'd explicitly extract the code `27` into a variable, because it is not clear what key are you listening to, e.g.
```
const ESC = 27;
if (e.keyCode === ESC) {}
```